src/main/kotlin/uk/gov/justice/digital/hmpps/notificationsalertsvsip/service/EmailSenderService.kt

```diff
@@ -64,31 +73,77 @@ class EmailSenderService(
   }
 
   private fun handleBookedEvent(visit: VisitDto): SendEmailNotificationDto {
-    val templateVars = mutableMapOf(
+    LOG.info("handleBookedEvent (email) - Entered")
+
+    val templateVars = getCommonTemplateVars(visit)
+
+    templateVars.putAll(
+      mutableMapOf(
+        "time" to dateUtils.getFormattedTime(visit.startTimestamp.toLocalTime()),
+        "end time" to dateUtils.getFormattedTime(visit.endTimestamp.toLocalTime()),
+        "arrival time" to "45",
+        "closed visit" to (visit.visitRestriction == VisitRestriction.CLOSED).toString(),
+        "visitors" to getVisitors(visit),
+      ),
+    )
+
+    return SendEmailNotificationDto(templateName = EmailTemplateNames.VISIT_BOOKING, templateVars = templateVars)
+  }
+
+  private fun handleCancelledEvent(visit: VisitDto): SendEmailNotificationDto {
+    LOG.info("handleCancelledEvent (email) - Entered")
+
+    return SendEmailNotificationDto(
+      templateName = getCancelledEmailTemplateName(visit.outcomeStatus!!),
+      templateVars = getCommonTemplateVars(visit),
+    )
+  }
+
+  private fun getCancelledEmailTemplateName(visitOutcome: OutcomeStatus): EmailTemplateNames {
+    return when (visitOutcome) {
+      OutcomeStatus.PRISONER_CANCELLED -> {
+        EmailTemplateNames.VISIT_CANCELLED_BY_PRISONER
+      }
+
+      OutcomeStatus.ESTABLISHMENT_CANCELLED, OutcomeStatus.DETAILS_CHANGED_AFTER_BOOKING, OutcomeStatus.ADMINISTRATIVE_ERROR -> {
+        EmailTemplateNames.VISIT_CANCELLED_BY_PRISON
+      }
+
+      OutcomeStatus.VISITOR_CANCELLED -> {
+        EmailTemplateNames.VISIT_CANCELLED
+      }
+
+      else -> {
+        LOG.error("visit cancellation type $visitOutcome is unsupported")
+        throw ValidationException("visit cancellation type $visitOutcome is unsupported")
```

Comment:
Spoke with Ed about other cancellation reasons. If public booker manages to send in a reason other than the ones supplied above, it's an error. So i've thrown a ValidationException in this case as it shouldn't have been received.